### PR TITLE
Added system purpose data to inventory payload

### PR DESCRIPTION
**Pull request by mstead**
Pull system purpose attributes from pinhead data and send them
to the inventory service to be stored in the RHSM fact namespace

### CODE REVIEW

File: src/main/java/org/candlepin/insights/inventory/InventoryService.java [hunk 1]
```diff
@@ -73,6 +73,15 @@ protected CreateHostIn createHost(ConduitFacts conduitFacts, OffsetDateTime sync
         if (conduitFacts.getRhProd() != null) {
             rhsmFactMap.put("RH_PROD", conduitFacts.getRhProd());
         }
+        if (conduitFacts.getSysPurposeRole() != null && !conduitFacts.getSysPurposeRole().isEmpty()) {
+            rhsmFactMap.put("SYS_PURPOSE_ROLE", conduitFacts.getSysPurposeRole());
```
Comment (kahowell) [inline src/main/java/org/candlepin/insights/inventory/InventoryService.java]: Can we please rename to SYSPURPOSE_ROLE for consistency w/ RedHatInsights/rhsm-subscriptions#48 ?

File: src/main/java/org/candlepin/insights/inventory/InventoryService.java [hunk 2]
```diff
@@ -73,6 +73,15 @@ protected CreateHostIn createHost(ConduitFacts conduitFacts, OffsetDateTime sync
...
+        }
+        if (conduitFacts.getSysPurposeUsage() != null && !conduitFacts.getSysPurposeUsage().isEmpty()) {
+            rhsmFactMap.put("SYS_PURPOSE_USAGE", conduitFacts.getSysPurposeUsage());
```
Comment (kahowell) [inline src/main/java/org/candlepin/insights/inventory/InventoryService.java]: Can we please rename to SYSPURPOSE_USAGE for consistency w/ RedHatInsights/rhsm-subscriptions#48 ?

File: src/main/java/org/candlepin/insights/inventory/InventoryService.java [hunk 3]
```diff
@@ -73,6 +73,15 @@ protected CreateHostIn createHost(ConduitFacts conduitFacts, OffsetDateTime sync
...
+        }
+        if (conduitFacts.getSysPurposeAddons() != null) {
+            rhsmFactMap.put("SYS_PURPOSE_ADDONS", conduitFacts.getSysPurposeAddons());
```
Comment (kahowell) [inline src/main/java/org/candlepin/insights/inventory/InventoryService.java]: Can we please rename to SYSPURPOSE_ADDONS for consistency w/ RedHatInsights/rhsm-subscriptions#48 ?

File: src/main/java/org/candlepin/insights/inventory/InventoryService.java (hunk 1 above)
Comment (mstead) [reply src/main/java/org/candlepin/insights/inventory/InventoryService.java]: DONE

File: src/main/java/org/candlepin/insights/inventory/InventoryService.java (hunk 2 above)
Comment (mstead) [reply src/main/java/org/candlepin/insights/inventory/InventoryService.java]: DONE

File: src/main/java/org/candlepin/insights/inventory/InventoryService.java (hunk 3 above)
Comment (mstead) [reply src/main/java/org/candlepin/insights/inventory/InventoryService.java]: DONE

**Comment by mstead** [comment]
@kahowell Requested changes addressed.

**Pull request merged**